### ISSUE TITLE
Glossary + "anatomy of a figure" page?

### ISSUE THREAD

**Issue by jariji** [MakieOrg/Makie.jl#2729]
Just an idea.

Makie uses a lot of terms in Makie-specific ways and I find it hard to remember how they're all related. It might be helpful to have a doc page with a glossary defining all the different things. Scene, Figure, Layout, Plot, Axes, axis, Observable, ...

And perhaps a diagram showing the components. Matplotlib gives a [diagram](https://matplotlib.org/stable/gallery/showcase/anatomy.html) showing the components in its model. 
![image](https://user-images.githubusercontent.com/96840304/222984477-5a6b21e9-fa92-49d8-a9bb-a3dee59333f4.png)

**Comment by jkrumbiegel** [comment]
I agree, would be good to have. In Makie, it's a little less split into components, in matplotlib basically everything is a class but in `Axis`, for example the grid is just a couple of line primitives derived from the tick position and axis area observables
